modules/internal/DefaultRectangular.chpl

```diff
@@ -295,72 +295,44 @@ module DefaultRectangular {
                                                      minIndicesPerTask,
                                                      ranges);
-      if debugDefaultDist {
-        chpl_debug_writeln("    numChunks=", numChunks, " parDim=", parDim,
-                " ranges(", parDim, ").length=", ranges(parDim).length);
-      }
-
-      if debugDataPar {
-        chpl_debug_writeln("### numTasksPerLoc = ", numTasks, "\n" +
-                "### ignoreRunning = ", ignoreRunning, "\n" +
-                "### minIndicesPerTask = ", minIndicesPerTask, "\n" +
-                "### numChunks = ", numChunks, " (parDim = ", parDim, ")\n" +
-                "### nranges = ", ranges);
-      }
-
       if numChunks <= 1 {
         for i in these_help(1) {
           yield i;
         }
       } else {
-        var locBlock: rank*range(intIdxType);
-        for param i in 1..rank {
-          locBlock(i) = offset(i)..#(ranges(i).length);
-        }
         if debugDefaultDist {
-          chpl_debug_writeln("*** DI: locBlock = ", locBlock);
+          chpl_debug_writeln("    numChunks=", numChunks, " parDim=", parDim,
+                             " ranges(", parDim, ").length=", ranges(parDim).length);
+          chpl_debug_writeln("### numTasksPerLoc = ", numTasks, "\n" +
+                             "### ignoreRunning = ", ignoreRunning, "\n" +
+                             "### minIndicesPerTask = ", minIndicesPerTask, "\n" +
+                             "### numChunks = ", numChunks, " (parDim = ", parDim, ")\n");
+          chpl_debug_writeln("*** DI: ranges = ", ranges);
         }
+        // TODO: The following is somewhat of an abuse of what
+        // _computeBlock() was designed for (dense ranges only; I
+        // multiplied by the stride as a white lie to make it work
+        // reasonabley.  We should switch to using the RangeChunk
+        // library...
         coforall chunk in 0..#numChunks {
-          var followMe: rank*range(intIdxType) = locBlock;
-          const (lo,hi) = _computeBlock(locBlock(parDim).length,
+          var block = ranges;
+          const len = if (!ranges(parDim).stridable) then ranges(parDim).length
+              else ranges(parDim).length:uint * abs(ranges(parDim).stride):uint;
+          const (lo,hi) = _computeBlock(len,
                                         numChunks, chunk,
-                                        locBlock(parDim)._high,
-                                        locBlock(parDim)._low,
-                                        locBlock(parDim)._low);
-          followMe(parDim) = lo..hi;
-          if debugDefaultDist {
-            chpl_debug_writeln("*** DI[", chunk, "]: followMe = ", followMe);
+                                        ranges(parDim)._high,
+                                        ranges(parDim)._low,
+                                        ranges(parDim)._low);
+          if (block(parDim).stridable) then
```

Comment:
nit -- odd mix of `then` and braces in the else branch. paren around condition also seems a little out of place

Reply:
Fixed in PR #13308.